### PR TITLE
feat: added filters

### PR DESCRIPTION
**Pull request by AtulHande03a**
-all required filter ie sortBy price, category, rating have been added
- redirecting to product page with selected category from home page has been added.

### CODE REVIEW

**Review by vijaytembugade — APPROVED** [review]
Good to go. 
Suggestion: Give min-height to the card

File: src/components/Categories.jsx [hunk 1]
```diff
@@ -1,33 +1,58 @@
-// import { categories } from "data.js";
-import { Link } from "react-router-dom";
+import React, { useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+// import { Link } from "react-router-dom";
```
Comment (t007rushi) [inline src/components/Categories.jsx:3]: remove comments

File: src/components/Filters.jsx [hunk 2]
```diff
@@ -1,13 +1,50 @@
+import { useFilter } from "contexts/filter-context";
 import React from "react";
 
 export const Filters = () => {
+  const { filterState, filterDispatch } = useFilter();
+
+  const { price, sortBy, category, rating } = filterState;
+  const { sofa, chair, table, beds } = category;
+
+  const priceHandler = (e) => {
```
Comment (t007rushi) [inline src/components/Filters.jsx:10]: nice that you extracted out things

File: src/components/Products.jsx [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+import { filterProducts } from "utils/filterProducts";
 import { ProductCard } from "./ProductCard";
 
 export const Products = () => {
   const { loading, productList } = useProducts();
+  const { filterState } = useFilter();
+  const sortedProducts = filterProducts(productList, filterState);
+  console.log(productList, "productlist");
+  console.log(sortedProducts, "sortedproducts");
+  console.log(filterProducts(productList, filterState), "func");
```
Comment (t007rushi) [inline src/components/Products.jsx:13]: remove cosoles

**Review by t007rushi — APPROVED** [review]
Liked the theme Greta go-ahead

**Pull request merged**